packages/next/shared/lib/is-plain-object.ts

```diff
@@ -8,5 +8,5 @@ export function isPlainObject(value: any): boolean {
   }
 
   const prototype = Object.getPrototypeOf(value)
-  return prototype === null || prototype === Object.prototype
+  return prototype === null || prototype.hasOwnProperty('isPrototypeOf')
```

Comment:
We add a comment for why we switch to this because of the `Object.prototype` is now bit different in context

Reply:
done!